### ISSUE TITLE
Disable the selection of all columns，cause drag the nested header structure error.

### ISSUE THREAD

**Issue by hutiantian04** [handsontable/handsontable#6026]
### Description

Disable the selection of all columns in the beforeOnCellMouseDown event.
### Steps to reproduce

1.
  beforeOnCellMouseDown: function(e, coords, td){
      if (coords.row < 0) {

        event.stopImmediatePropagation();

    }
  },
2.Create a nested header structure
3.Drag the head
![image](https://user-images.githubusercontent.com/39802580/58163415-8e9c2280-7cb6-11e9-83cc-c782c5182e58.png)

### Demo

http://jsrun.net/EXyKp/edit

### Your environment
* Handsontable version:7.0.3
* Browser Name and version:Google Chrome version: 73.0.3683.86
* Operating System:win10

**Comment by AMBudnik** [comment]
Hey @hutiantian04 

how do you `Drag the head`? Can you record it?

**Comment by hutiantian04** [comment]
Hi AMBudnik
Sorry, I didn't say clearly about the drag. 
Mouse click on a column in the head, hold down the mouse and don't loosen it, then drag it.
![webwxgetmsgimg](https://user-images.githubusercontent.com/39802580/58223787-abd00000-7d4d-11e9-8917-9fc8844555d0.gif)

**Comment by AMBudnik** [comment]
You're right the `beforeOnCellMouseDown` hook produces the errors. It happens even when there is only one row of headers https://jsfiddle.net/AMBudnik/b6pnfsko/

**Comment by AMBudnik** [comment]
The issue is no longer replicable in v 7.3.0 https://jsfiddle.net/yv04wdm7/ but previously I was on Windows 10 and now I'm on macOS Catalina so I would be grateful for feedback. Could you check if 7.3.0 works for you @hutiantian04 ?

**Comment by AMBudnik** [comment]
Works the same in `7.4.2`, `8.0.0-beta1`. We just need a simple selection.

![error](https://user-images.githubusercontent.com/10757813/77070970-85213b00-69eb-11ea-9732-33397d8d0152.gif)

**Comment by pnowak** [comment]
This issue is still replicable using v8.0.0. Here's an updated demo - https://jsfiddle.net/4atoxjfs/

**Comment by AMBudnik** [comment]
This issue is still replicable using v8.2.0 https://jsfiddle.net/handsoncode/1hLnecjd/

**Comment by budnix** [comment]
Thanks for reporting. The issue is fixed in https://github.com/handsontable/handsontable/pull/12794.
